src/utils.js

```diff
@@ -1077,11 +1077,57 @@ go.utils = {
             && im.user.state.name !== 'states_start';
     },
 
-    opt_out: function(im, contact, optout_reason, api_optout, unsub_all, jembi_optout, metric_prefix) {
+    get_reg_source: function(contact) {
+        var reg_source;
+        var reg_options = ['clinic', 'chw', 'personal'];
+        if (!_.contains(reg_options, contact.extra.is_registered_by)) {
+            reg_source = 'unknown';
+        } else {
+            reg_source = contact.extra.is_registered_by;
+        }
+        return reg_source;
+    },
+
+    adjust_percentage_optouts: function(im, env) {
+        var m_store = im.config.metric_store;
+        return Q.all([
+            go.utils.get_kv(im, [m_store, env, 'sum', 'subscriptions'].join('.'), 0),
+            go.utils.get_kv(im, [m_store, env, 'sum', 'optouts'].join('.'), 0),
+            go.utils.get_kv(im, [m_store, env, 'sum', 'optout_cause', 'non_loss'].join('.'), 0),
+            go.utils.get_kv(im, [m_store, env, 'sum', 'optout_cause', 'loss'].join('.'), 0),
+            go.utils.get_kv(im, [m_store, env, 'optout', 'sum', 'subscription_to_protocol_success'].join('.'), 0)
+        ]).spread(function(total_subscriptions, total_optouts, non_loss_optouts, loss_optouts, loss_msg_signups) {
+            var percentage_optouts = parseFloat( ((total_optouts/total_subscriptions)*100).toFixed(2) );
+            var percentage_non_loss_optouts = parseFloat(((non_loss_optouts / total_subscriptions) * 100).toFixed(2));
+            var percentage_loss_msg_signups = parseFloat(((loss_msg_signups / loss_optouts) * 100).toFixed(2));
+            return Q.all([
+                im.metrics.fire.last([env, 'percent', 'optout', 'all'].join('.'), percentage_optouts),
+                im.metrics.fire.last([env, 'percent', 'optout', 'non_loss'].join('.'), percentage_non_loss_optouts),
+                im.metrics.fire.last([env, 'percent', 'optout', 'loss', 'msgs'].join('.'), percentage_loss_msg_signups)
```

Comment:
Yes, I did it on purpose, thinking it differentiate four metrics (which we might need in the future / should maybe add here):
1. `non_loss`: all non loss optouts
2. `loss.msgs`: loss optouts that chose to get messages
3. `loss.no_msgs`: loss optouts that chose not to get messages
4. `loss`: sum of 2 & 3
But let me know if you'd like me to change it.